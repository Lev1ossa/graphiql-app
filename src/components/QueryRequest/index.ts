export { default as QueryRequest } from './QueryRequest'
