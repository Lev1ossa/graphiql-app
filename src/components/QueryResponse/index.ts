export { default as QueryResponse } from './QueryResponse'
