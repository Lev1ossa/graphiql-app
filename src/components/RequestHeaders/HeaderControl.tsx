import { PredefinedHeaders } from '@/constants/constants'
import { useLocale } from '@/hooks/useLocale'
import { Delete } from '@mui/icons-material'
import {
  Autocomplete,
  Box,
  Checkbox,
  IconButton,
  TextField,
} from '@mui/material'

export default function HeaderControl(props: {
  checked: boolean
  headerKey: string
  headerVal: string
  handleCheck: (checked: boolean) => void
  handleKeyChange: (key: string) => void
  handleValChange: (value: string) => void
  handleDelete: () => void
}) {
  const {
    checked,
    headerKey,
    headerVal,
    handleCheck,
    handleKeyChange,
    handleValChange,
    handleDelete,
  } = props

  const { locale } = useLocale()

  return (
    <Box
      sx={{
        display: 'grid',
        gap: 1,
        gridTemplateAreas: `"check key value delete"`,
        gridTemplateColumns: 'max-content 1fr 1fr max-content',
        justifyItems: 'center',
        '@container (max-width: 400px)': {
          gridTemplateAreas: `
            "key   key   check"
            "value value delete"
          `,
          gridTemplateColumns: '1fr max-content',
        },
      }}
    >
      <Box sx={{ gridArea: 'check' }}>
        <Checkbox
          value={checked}
          sx={{
            '&.Mui-disabled': {
              cursor: 'not-allowed',
              pointerEvents: 'painted',
            },
          }}
          disabled={!headerKey.length}
          onChange={(_, checked) => handleCheck(checked)}
        />
      </Box>
      <Autocomplete
        sx={{ gridArea: 'key' }}
        size="small"
        freeSolo
        fullWidth
        options={PredefinedHeaders}
        renderInput={(params) => (
          <TextField {...params} label={locale.mainPage.inputLabel.headerKey} />
        )}
        value={headerKey}
        onChange={(_, value) => handleKeyChange(value || '')}
      />
      <TextField
        sx={{ gridArea: 'value' }}
        size="small"
        fullWidth
        label={locale.mainPage.inputLabel.headerValue}
        value={headerVal}
        onChange={(event) => handleValChange(event.currentTarget.value)}
      />
      <Box sx={{ gridArea: 'delete' }}>
        <IconButton color="error" onClick={() => handleDelete()}>
          <Delete />
        </IconButton>
      </Box>
    </Box>
  )
}
