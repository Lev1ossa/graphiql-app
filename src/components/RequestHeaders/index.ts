export { default as RequestHeaders } from './RequestHeaders'
