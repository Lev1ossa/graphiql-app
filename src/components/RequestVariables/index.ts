export { default as RequestVariables } from './RequestVariables'
