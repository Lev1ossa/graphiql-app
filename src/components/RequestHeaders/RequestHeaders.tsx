import { useLocale } from '@/hooks/useLocale'
import { useBoundStore } from '@/store'
import { Add } from '@mui/icons-material'
import { Box, Button, Stack } from '@mui/material'
import HeaderControl from './HeaderControl'

export default function RequestHeaders() {
  const headers = useBoundStore((state) => state.headers)
  const createHeader = useBoundStore((state) => state.createHeader)
  const deleteHeader = useBoundStore((state) => state.deleteHeader)
  const changeHeader = useBoundStore((state) => state.changeHeader)

  const {
    locale: { mainPage },
  } = useLocale()

  return (
    <Stack
      height={1}
      sx={{
        overflow: 'overlay',
        padding: 2,
        boxSizing: 'border-box',
        container: 'headers / inline-size',
      }}
    >
      <Stack gap={1}>
        {headers.map(({ id, checked, headerKey, headerVal }) => (
          <HeaderControl
            key={id}
            checked={checked}
            headerKey={headerKey}
            headerVal={headerVal}
            handleCheck={(checked) => changeHeader(id, { checked })}
            handleKeyChange={(headerKey) => changeHeader(id, { headerKey })}
            handleValChange={(headerVal) => changeHeader(id, { headerVal })}
            handleDelete={() => deleteHeader(id)}
          />
        ))}
      </Stack>
      <Box alignSelf="end" mt={2} pr={1}>
        <Button
          startIcon={<Add />}
          variant="outlined"
          onClick={() => createHeader(crypto.randomUUID())}
        >
          {mainPage.button.addHeader}
        </Button>
      </Box>
    </Stack>
  )
}
