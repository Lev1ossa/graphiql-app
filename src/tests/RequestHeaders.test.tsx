import { RequestHeaders } from '@/components/RequestHeaders'
import HeaderControl from '@/components/RequestHeaders/HeaderControl'
import { describe, expect, test } from 'vitest'
import renderInContext from './renderInContext'

describe('Correct RequestHeaders snapshot', () => {
  test('Correct [RequestHeaders] snapshot', () => {
    const { baseElement } = renderInContext(<RequestHeaders />)
    expect(baseElement).toMatchSnapshot()
  })
  test('Correct HeaderControl snapshot', () => {
    const props = {
      handleCheck: () => {},
      handleDelete: () => {},
      handleKeyChange: () => {},
      handleValChange: () => {},
      headerKey: '',
      headerVal: '',
    }
    expect(<HeaderControl checked={false} {...props} />).toMatchSnapshot()
    expect(<HeaderControl checked={true} {...props} />).toMatchSnapshot()
  })
})
