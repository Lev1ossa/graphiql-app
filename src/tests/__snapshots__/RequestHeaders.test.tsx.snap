// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct RequestHeaders snapshot > Correct [RequestHeaders] snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiStack-root css-1cfeh5m-MuiStack-root"
    >
      <div
        class="MuiStack-root css-hwnj0i-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rt5s2b"
        >
          <div
            class="MuiBox-root css-1ac9ip6"
          >
            <span
              aria-disabled="true"
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-53bw56-MuiButtonBase-root-MuiCheckbox-root"
              tabindex="-1"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                disabled=""
                type="checkbox"
                value="false"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </div>
          <div
            class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon css-1nsar26-MuiAutocomplete-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Header key
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  role="combobox"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                    tabindex="-1"
                    title="Clear"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="CloseIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Header key
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-120l7bi-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Value
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Value
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-1x254sl"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1tz72dr"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          add header
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Correct RequestHeaders snapshot > Correct HeaderControl snapshot 1`] = `
<HeaderControl
  checked={false}
  handleCheck={[Function]}
  handleDelete={[Function]}
  handleKeyChange={[Function]}
  handleValChange={[Function]}
  headerKey=""
  headerVal=""
/>
`;

exports[`Correct RequestHeaders snapshot > Correct HeaderControl snapshot 2`] = `
<HeaderControl
  checked={true}
  handleCheck={[Function]}
  handleDelete={[Function]}
  handleKeyChange={[Function]}
  handleValChange={[Function]}
  headerKey=""
  headerVal=""
/>
`;
